### PR TITLE
update ShortGrainDropper

### PR DESCRIPTION
**Pull request by wangyems**

### CODE REVIEW

File: src/Featurizers/ShortGrainDropperFeaturizer.h [hunk 1]
```diff
@@ -85,13 +85,13 @@ class ShortGrainDropperEstimator : public TransformerEstimator<std::vector<std::
     ShortGrainDropperEstimator(
         AnnotationMapsPtr pAllColumnAnnotations,
         size_t colIndex,
-        std::uint8_t windowSize,
+        std::uint32_t maxWindowSize,
         //todo: possible name change and add commments, after sync with other Timeseries related Featurizers
```
Comment (davidbrownellWork) [inline src/Featurizers/ShortGrainDropperFeaturizer.h]: >//todo: possible name change and add commments, after sync with other Timeseries related Featurizers [](start = 8, length = 100)

Remove these todo comments.

File: src/Featurizers/ShortGrainDropperFeaturizer.h [hunk 2]
```diff
@@ -192,9 +192,9 @@ ShortGrainDropperEstimator<MaxNumTrainingItemsV>::ShortGrainDropperEstimator(
             //cv may not be 0, not sure currently
```
Comment (davidbrownellWork) [inline src/Featurizers/ShortGrainDropperFeaturizer.h]: Update or remove.

File: src/Featurizers/ShortGrainDropperFeaturizer.h [hunk 3]
```diff
@@ -192,9 +192,9 @@ ShortGrainDropperEstimator<MaxNumTrainingItemsV>::ShortGrainDropperEstimator(
...
             if (std::distance(std::get<0>(lags), std::get<1>(lags)) == 0)
                 throw std::invalid_argument("lags");
-            if (!cv.has_value())
-                return (maxHorizon + std::max(windowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
-            return (2*maxHorizon + static_cast<std::uint8_t>(*cv) + std::max(windowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
+            if (!crossValidation.has_value())
+                return (horizon + std::max(maxWindowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
+            return (2*horizon + *crossValidation + std::max(maxWindowSize, *std::max_element(std::get<0>(lags), std::get<1>(lags))) + 1);
```
Comment (davidbrownellWork) [inline src/Featurizers/ShortGrainDropperFeaturizer.h]: I don't understand what this code is doing. Jin and I have spent a lot of time talking about a similar topic with regards to the LagLeadOperator featurizer. Can you sync with him and try to make this code simpler.

Comment (wangyems) [reply src/Featurizers/ShortGrainDropperFeaturizer.h]: sure

**Comment by davidbrownellWork** [comment]
>     _groupByGrains = {};

Could this work be done in `create_transformer_impl`? If so, you wouldn't need to keep a `_grainsToDrop` member variable as it could be local to that func.

---
Refers to: src/Featurizers/ShortGrainDropperFeaturizer.h:194 in 405aaca. [](commit_id = 405aaca86010903de8b8b5fc84a65d638d682784, deletion_comment = False)

**Comment by davidbrownellWork** [comment]
>     using GrainsSet                         = ShortGrainDropperTransformer::GrainsSet;

If you end up removed `_grainsToDrop`, remove this as well. In the implementation of `create_transformer_impl`, use the type defined in `ShortGrainDropperTransformer`.

---
Refers to: src/Featurizers/ShortGrainDropperFeaturizer.h:97 in 405aaca. [](commit_id = 405aaca86010903de8b8b5fc84a65d638d682784, deletion_comment = False)

**Pull request merged**